### ISSUE TITLE
Please support +lds flag for vectory memory operations in ROCm

### ISSUE THREAD

**Issue by 949f45ac** [ROCm/ROCm#630]
In section 12.18, the [Vega ISA guide](https://developer.amd.com/wp-content/resource/Vega_Shader_ISA_28July2017.pdf) states that vector memory operations support a flag called `lds` with the following effect:
>Data is transferred between LDS and Memory, not VGPRs.

I am interested in trying this out, but neither is this flag ever generated by the compiler (for example it will issue `global_load` followed by `ds_write` instead of `global_load … lds`) nor accepted when I write inline asm. I am using HIP/hcc.

Thanks!

**Comment by 949f45ac** [comment]
Also it would be really great if there were intrinsics that allowed defining these flags (glc, slc) for vector memory operations, without the need to issue asm directly. This is because doing these via asm _efficiently_ is quite complicated, for you have to issue the `s_waitcnt` manually, too, and at the ideal position. (The compiler will place the instruction and the `s_waitcnt` as far apart from one another as possible, to give the operation maximal time to run in parallel. Hard to imitate by hand.)

**Comment by preda** [comment]
If clinfo reports the cl_amd_assembly_program extension, you can feed a full kernel in assembly. Although this is complicated unless the kernel is tiny, and there is not much documentation around cl_amd_assembly_program or examples of its use.

**Comment by preda** [comment]
Also, if you could show one case where your hand-rolled code using direct LDS  - global memory is faster then what the compiler generates, you'd have a strong case for a compiler performance bug.

**Comment by tasso** [comment]
Is this still an issue? If not, can we please close it?

**Comment by tasso** [comment]
Original ticket is more than a year old and the person that opened ticket originally has not responded to the latest request.  If this is still an issue, please file a new ticket and we will be more than happy to investigate it.  Thanks!
